### PR TITLE
Updated setImage to setImageUrl for Improved API and cancelling logic

### PR DESCRIPTION
**Pull request by gugges**

### CODE REVIEW

File: Example/Example/ViewController.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ final class ViewController: UIViewController {
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         
-        imageView.ok.setImage(with: URL(string: "https://www.gstatic.com/webp/gallery/4.webp")!)
+        imageView.ok.setImageUrl(with: URL(string: "https://www.gstatic.com/webp/gallery/4.webp")!)
```
Comment (Dmitri-Fuerle-okcupid) [inline Example/Example/ViewController.swift]: this doesn't read right to me. Either `imageView.ok.setImageUrl(URL(string:""))` or `imageView.ok.setImage(url: URL(string:""))`

Comment (gugges) [reply Example/Example/ViewController.swift]: Ok, update to url. I don't like omitting the var name as that is less standard in swift syntax

File: Example/Example/ViewController.swift [hunk 2]
```diff
@@ -15,7 +15,7 @@ final class ViewController: UIViewController {
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         
-        imageView.ok.setImage(with: URL(string: "https://www.gstatic.com/webp/gallery/4.webp")!)
+        imageView.ok.setImageUrl(url: URL(string: "https://www.gstatic.com/webp/gallery/4.webp")!)
```
Comment (Dmitri-Fuerle-okcupid) [inline Example/Example/ViewController.swift]: you are writing url twice here

File: Sources/Extensions/UIButton+ImageDownloader.swift [hunk 3]
```diff
@@ -9,11 +9,25 @@
...
+        set {
+            setImageUrl(url: newValue)
+        }
+    }
     
-    func setImage(with url: URL,
-                  for state: UIControl.State = .normal,
-                  imageDownloader: ImageDownloading = ImageDownloader.shared,
-                  completionHandler: ImageDownloader.CompletionHandler? = nil) {
+    func setImageUrl(url: URL?,
```
Comment (Dmitri-Fuerle-okcupid) [inline Sources/Extensions/UIButton+ImageDownloader.swift]: make it either `setImage(url: URL` or `setImageURL(_ url: URL)`

Comment (gugges) [reply Sources/Extensions/UIButton+ImageDownloader.swift]: ok done

File: Sources/Extensions/UIButton+ImageDownloader.swift [hunk 4]
```diff
@@ -9,11 +9,25 @@
 import UIKit
 
 public extension ObjectWrapper where T: UIButton {
```
Comment (Dmitri-Fuerle-okcupid) [inline Sources/Extensions/UIButton+ImageDownloader.swift:11]: Should this be supported? With a button I could have multiple images one for each state and this doesn't work with that. maybe it should be removed? do you use it?

Comment (gugges) [reply Sources/Extensions/UIButton+ImageDownloader.swift:11]: The setImage function does take a state argument and can properly set different images.

The imageUrl property is a convenience that will set the normal state by default.

Let me know if you think we should remove the imageUrl convenience property.

**Review by Dmitri-Fuerle-okcupid — APPROVED** [review]
LGTM

**Pull request merged**